doc/design/csp/select_op.md

```diff
@@ -0,0 +1,94 @@
+# select_op Design
+
+## Introduction
+
+In golang, the **select** statement lets a goroutine wait on multiple 
+communication operations at the same time. The **select** blocks until
+one of its cases can run, then executes the case.  If multiple cases
+are ready to run, then one case is choosen at random to be executed.
+
+With the introduction of CSP for Paddle, we mimic this behavior by 
+creating a ***select_op***.  The ***select_op*** works in conjunction
+with the **channel_send_op**, **channel_receive_op**, and 
+**channel_close_op**.
+
+## How to use it
+
+The **select_op** is available as a c++ operator.  However most users
+will prefer to use the much simplier Python API.
+
+- **fluid.Select()**: Creates a Select block and adds it to the main
+program block.  Within the select block, users can add cases by calling 
+**select.case** or **select.default** method.
+
+- **fluid.Select.case(channel_action, channel, result_variable)**: Represents
+a fluid channel send/recv case.  This method creates a SelectCase block
+guard and adds it to the Select block.  The arguments into this method tells
+the select which channel operation to listen to.
+
+- **fluid.Select.default()**: Represents the fluid default case.  This default
+case is executed if none of the channel send/recv cases are available to
+execute.
+
+**Example:**
+```
+x = fill_constant(shape=[1], dtype=core.VarDesc.VarType.INT32, value=0)
+y = fill_constant(shape=[1], dtype=core.VarDesc.VarType.INT32, value=1)
+ 
+while_cond = fill_constant(shape=[1], dtype=core.VarDesc.VarType.BOOL, value=True)
+while_op = While(cond=while_cond)    
+ 
+with while_op.block():
```

Comment:
While is a little tricky case - not sure I would use it for an example introducing it for the first time. BUT if you want to use to show how it can work with while, you might have to add a little more text perhaps below the example to explain why we do that funky `append_op` business to get out of the `while`. Because there is nothing normal about it.